docs/providers.rst

```diff
@@ -142,6 +142,12 @@ In Social App:
 page, or it’s your App ID Prefix in your App ID)
     Cert: The contents of your private key, including "-----BEGIN/END PRIVATE KEY-----" and line breaks
 
+In your django settings, the means by which apple returns data to your app requires that cross-site cookies are allowed. This can be done in your Django settings:
+
+.. code-block:: python
+
+    SESSION_COOKIE_SAMESITE = None
```

Comment:
Is it «secure» to post this flag for web apps?

Reply:
Right, so I've looked in to this a bit, it should not be a problem since the setting only applies to the session cookie. If it was global then this would be bad (i.e. the csrf token must have at least "Lax").

I'm adding to the docs to say that `SESSION_COOKIE_SECURE = True` should also be set too

Reply:
Hmm, okay. But I am still not sure that this decision is better than redirect `get->post`. If you can, please convince me. 
Anyway, @pennersr has the last word.

Reply:
I am also thinking about the tradeoff. Maybe we can make the thing stateless for apple!

Reply:
I think we should implement a different approach for apple as they support [state using query params](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/incorporating_sign_in_with_apple_into_other_platforms#3332115) instead of session. We need to add the state according to it. Setting this value in settings may lead unintended security risk.

Reply:
So, a moderate problem with the previous approach was that the user's name is only provided in the `user` key of the`form_post` body. With the `POST`->`GET` redirect, there is nowhere to safely store the user's name.

Making this particular flow stateless would be a large deviation from the way that allauth works for all other providers at the moment. 

As mentioned, the `SESSION_COOKIE_SAMESITE` only applies to the session cookie, so it would not prevent django's CSRF protection, which to my mind severely limits the ability of this to cause problems, especially if CORS is correctly configured. 

That said, I am not a web security expert.

Reply:
I've created a new (and arguably better) approach here: https://github.com/0leg5ergeev/django-allauth/pull/9

The TL;DR is:
- Use a `POST`->`GET` redirect as the apple form_post return endpoint
- Create an apple login specific session (since the normal `request.session` is not available with default/secure cookie settings with a cross-origin post). This stores the `id_token` and `user` from the response. You can think of this endpoint as making the apple response behave more like a normal oauth2 response
- The subsequent endpoint is much more normal for allauth and fetches the data it needs back from the apple specific session (then removes the apple session)

It's not perfect, but it does avoid changing cookie settings to be potentially insecure and it also keeps the apple specific code very contained to this new endpoint.

Keen for feedback!